Add script going through whole routine

busykoala opened the pull request:
Close #16 since it should be implemented to search in `app_root/directory` as well as expected.

One should only have to execute one command and the whole routine starts.
Therefore this script goes through every task the linkchecker has to do.

Commit:
Add script going through whole routine

One should only have to execute one command and the whole routine starts.
Therefore this script goes through every task the linkchecker has to do.

maethu commented on ftw/linkchecker/command/checking_links.py:
```diff
@@ -1,35 +1,86 @@
 
-def main(plone, *args):
+def get_plone_sites_information(app):
+    plone_sites = [obj for obj in app.objectValues(
```
The plone sites are usually nested in to mount-points (directories). so this script should follow directories too. --> ask @Nachtalb

Nachtalb replied:
@busykoala here is an example of how I do it, if you have any questions, feel free to ask.

https://github.com/4teamwork/bgbern.ng/blob/dbc4612e77617727d27e2c275b9187b65804c5ca/bgbern/ng/utils.py#L65-L84

busykoala replied:
I adapted my function to the given example.
Should we, like it's done in the given example, later only get PloneSites without self? For developing reasons I'll leave that out now.

Please move the discussion to #16 so we can move on.

maethu commented on ftw/linkchecker/command/checking_links.py:
```diff
@@ -1,35 +1,86 @@
+
+
+def catalog_lookup(relation_catalog):
+    x = relation_catalog
```
not necessary and try to avoid ambiguous variable names. Using `relation_catalog` is just fine.

busykoala replied:
👍

maethu commented on ftw/linkchecker/command/checking_links.py:
```diff
@@ -1,35 +1,86 @@
+        app,
+        site_info['id']
+    )
+    catalog_lookup(relation_catalog)
```
Since you only need this method once, it may not be necessary to move this 2 lines into it's own function. But it'll might be better for testing.

busykoala replied:
I tried to move it out of the main function and keep it clean by splitting the code into little tasks.

maethu commented on ftw/linkchecker/command/checking_links.py:
```diff
@@ -1,35 +1,86 @@
+def main(app, *args):
+    plone_sites_information = get_plone_sites_information(app)
+    for site_info in plone_sites_information:
+        email_address = 'hugo.boss@4teamwork.ch'
```
Is it planned to put this in some sort of configuration file?

busykoala replied:
I opened Issue #15.
I also asked about that myself here: https://github.com/4teamwork/ftw.linkchecker/pull/14#discussion_r235780887

Commit:
Follow dirs while getting PloneSites

The plone sites are usually nested in to mount-points (directories).o this script should follow directories too.

Commit:
Pin down dependency

This is necessary to be able to bootstrap since they did
recent changes.

Commit:
Fix according to #17

Nachtalb commented on ftw/linkchecker/command/checking_links.py:
```diff
@@ -0,0 +1,94 @@
+import AccessControl
+
+
+def get_plone_sites_information(app):
```
I would `get_plone_sites` extract to a function outside of this one. This makes it possible to be accessed by other parts in the code. Right now you maybe do not use this accessibility, but perhaps a new feature in the future needs it and then it will be easier for the developer. If you extract it, IMO it would need to go into a `helpers.py` or `utils.py`.

busykoala replied:
I'll take care of that in #18, so we can proceed with this pull request.

Nachtalb commented on ftw/linkchecker/command/checking_links.py:
```diff
@@ -0,0 +1,94 @@
+
+    plone_sites = get_plone_sites(app)
+
+    plone_sites_information = [{
```
I do not see a good enough reason to iterate over every PloneSite and only return this 3 information. For another developer who sees an element of this list in another function eg: `get_broken_relations_in_plone_page(app, site_info)` he will not know what keys are available. If you'd just return a list of `PloneSite`'s (or generator as in `get_plone_sites`) and would call the `get_broken_relations_in_plone_page` function with `plone_site` instead of `site_info` everyone would know, that it is a `PloneSite` and has all the available information.

Also if you iterate over the generator `get_plone_sites` as in `main`, you'd speed up the initial call, because you do not iterate over every object but only search for the next object after you have completed your operations on the current one.

busykoala replied:
I'll take care of that in #18, so we can proceed with this pull request.
